### PR TITLE
twistedchecker should think a comment starts in the middle of a docstring.

### PR DESCRIPTION
**Pull request by adiroiban**
"""
#twisted checker things this is a comment and so
1. Comments should begin with one whitespace
2. The first letter of comment should be capitalized

This is particularly triggered with url fragments
@see U{https://example.com/test#fragment}
"""

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2013-05-31T18:41:29Z
 * owner: tom.prince
 * the launchpad url was https://bugs.launchpad.net/bugs/1186378

### CODE REVIEW

**Comment by adiroiban** [comment]
Based on the provided example I tried to extend the checker to ignore comments inside docstrings.

Please check and let me know if changes make sense.

Thanks!

File: twistedchecker/functionaltests/comments.py [hunk 1]
```diff
@@ -15,4 +15,25 @@
...
+Line with an epydoc C{#} markup.
+
+This is particularly triggered with url fragments
+@see U{https://example.com/test#fragment}
+"""
+
+
+class SomeClass(object):
+    """
+    #twisted checker things this is a comment.
```
Comment (hawkowl) [inline twistedchecker/functionaltests/comments.py]: things -> thinks

File: twistedchecker/functionaltests/comments.py [hunk 2]
```diff
@@ -15,4 +15,25 @@
...
+
+    Line with an epydoc C{#} markup.
+
+    Also triggered with url fragments U{https://example.com/test#fragment}
+    """
```
Comment (hawkowl) [inline twistedchecker/functionaltests/comments.py]: Could there be an extra line with a wrong comment after the docstring, to make sure that it works after the docstring checks have been turned off?

**Comment by hawkowl** [comment]
Other than those two comments, LGTM.

**Comment by adiroiban** [comment]
Many thanks for the review and suggesting a better test suite.
The previous patch was buggy. I have fixed the code and updated the tests.

-------

Also if the result file has no empty new line at the end the assertion fails with a confusing message like this one:

```
twistedchecker.functionaltests.comments
= Expected =        = Actual =
1:W9402             1:W9402
3:W9401             3:W9401
5:W9401             5:W9401
7:W9402             7:W9402
31:W9401            31:W9401
33:W9401            33:W9401
43:W9401            43:W9401

twistedchecker.test.test_functionaltests.FunctionalTests.test_twistedchecker_functionaltests_comments
```

this is due to an extra newline. To fix this, I have changed to code to strip the values before comparison.

-------

Please check new changes

Thanks!

File: twistedchecker/functionaltests/comments.py [hunk 3]
```diff
@@ -15,4 +15,29 @@
 #  x = x + 1
 
 # Make sure no error occur when checking an empty comment
-#  
\ No newline at end of file
+#
+
+"""
+#twisted checker things this is a comment and so
```
Comment (hawkowl) [inline twistedchecker/functionaltests/comments.py]: things -> thinks

**Comment by adiroiban** [comment]
Typo fixed.
Please review latest changes.

Thanks!

**Pull request merged**